### DISCUSSION TITLE
reinstall frimware?

### DISCUSSION THREAD

**Discussion by 1pYDca4d5TUYfb** [gpdaniels/spike-prime#39] [Q&A]
I have a hub that was interrupted during the last update and is no longer functioning properly.  It is stuck in an endless upgrade cycle.  It no longer connects to any device via bluetooth and every time it is plugged it to a pc via USB it asks to upgrade and then reports an error without any options.  

Can this tool be used to reinstall firmware?  I've successfully connected to the hub with rshell but I don't know what to try next.

**Comment by SpudGunMan** [comment]
put it in DFU mode and flash pybricks on it and it should be happy to flash back if you get stuck

**Comment by gpdaniels** [comment]
There's no tool in this repository to flash firmware, it just contains a lot of historic firmware files. It should be possible to use a dfu tool to flash one of the firmwares onto the hub, but it's probably easier using pybricks tooling for that.
